### PR TITLE
added user option to perform rh kernel update

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

### CODE REVIEW

File: teuthology/run.py [hunk 1]
```diff
@@ -249,9 +249,11 @@ def get_initial_tasks(lock, config, machine_type):
             {'internal.git_ignore_ssl': None},
             {'internal.setup_cdn_repo': None},
             {'internal.setup_base_repo': None},
-            {'internal.setup_additional_repo': None},
-            {'kernel.install_latest_rh_kernel': None}
+            {'internal.setup_additional_repo': None}
         ])
+        # update kernel
+        if config.get('redhat').get('kernel_update', False):
```
Comment (kshtsk) [inline teuthology/run.py:255]: what if 'redhat' is undefined?

**Comment by kshtsk** [comment]
ok to test

File: teuthology/run.py [hunk 2]
```diff
@@ -249,9 +249,11 @@ def get_initial_tasks(lock, config, machine_type):
             {'internal.git_ignore_ssl': None},
             {'internal.setup_cdn_repo': None},
             {'internal.setup_base_repo': None},
-            {'internal.setup_additional_repo': None},
-            {'kernel.install_latest_rh_kernel': None}
+            {'internal.setup_additional_repo': None}
         ])
+        # Install latest kernel task for redhat downstream runs
+        if config.get('redhat').get('install_latest_rh_kernel', False):
```
Comment (vasukulkarni) [inline teuthology/run.py:255]: In the email i saw a comment from @kshtsk  but not here, I think we need to handle when 'redhat' is none.

Comment (vasukulkarni) [reply teuthology/run.py:255]: well the diff is confusing, the outer block has check so its safe.

**Pull request merged**